### PR TITLE
feat: subscription endpoints

### LINKED ISSUE #81: Add Subscription APIs

### PR DESCRIPTION
**Pull request by mtwichel**
Closes #81

### CODE REVIEW

File: lib/src/square_api_client.dart [hunk 1]
```diff
@@ -253,4 +253,95 @@ abstract class SquareApiClient {
...
+  ///
+  /// You can set, modify, and clear the subscription field values.
+  @PUT('/v2/subscriptions/{subscriptionId}')
+  Future<UpdateSubscriptionResponse> updateSubscription({
+    @Path() required String subscriptionId,
+    @Body() required UpdateSubscriptionRequest request,
+  });
+
+  /// Deletes a scheduled action for a subscription.
+  @DELETE('/v2/subscriptions/{subscriptionId}/actions{actionId}')
```
Comment (BruceClark406) [inline lib/src/square_api_client.dart:303]:
```suggestion
  @DELETE('/v2/subscriptions/{subscriptionId}/actions/{actionId}')
```

File: lib/src/square_api_client.dart [hunk 2]
```diff
@@ -253,4 +253,95 @@ abstract class SquareApiClient {
...
+  /// subscription.
+  @POST('/v2/subscriptions/{subscriptionId}/resume')
+  Future<ResumeSubscriptionResponse> resumeSubscription({
+    @Path() required String subscriptionId,
+    @Body() required ResumeSubscriptionRequest request,
+  });
+
+  /// Schedules a `SWAP_PLAN` action to swap a subscription plan in an
+  /// existing subscription.
+  @POST('/v2//subscriptions/{subscriptionId}/swap-plan')
```
Comment (BruceClark406) [inline lib/src/square_api_client.dart:342]:
```suggestion
  @POST('/v2/subscriptions/{subscriptionId}/swap-plan')
```

**Review by BruceClark406 — COMMENTED** [review]
A few comments, over all GREAT!
Where you going to clean up createPaymentLink in a different PR?

**Comment by mtwichel** [comment]
> Where you going to clean up createPaymentLink in a different PR?

@BruceClark406 I totally forgot about that, but yeah I'll do a separate PR

**Review by BruceClark406 — APPROVED** [review]

**Pull request merged**